pyamg/aggregation/smooth.py

```diff
@@ -1136,12 +1222,18 @@ def energy_prolongation_smoother(A, T, Atilde, B, Bf, Cpt_params,
     if Cpt_params[0]:
         Sparsity_Pattern = Cpt_params[1]['I_F'] * Sparsity_Pattern
         Sparsity_Pattern = Cpt_params[1]['P_I'] + Sparsity_Pattern
+        cost[0] += Sparsity_Pattern.nnz/float(A.nnz) 
 
     # Construct array of inv(Bi'Bi), where Bi is B restricted to row i's
     # sparsity pattern in Sparsity Pattern. This array is used multiple times
     # in Satisfy_Constraints(...).
     BtBinv = compute_BtBinv(B, Sparsity_Pattern)
```

Comment:
I would move this cost estimate into compute_BtBinv, i.e., make cost=[0.0] a parameter to compute_BtBinv